Mathlib/RingTheory/Algebraic/Basic.lean

```diff
@@ -560,6 +560,15 @@ theorem IsAlgebraic.exists_nonzero_dvd
   rw [map_sub, hq, zero_sub, dvd_neg, aeval_X, aeval_C] at key
   exact ⟨q.coeff 0, hq0, key⟩
 
+open Function (Injective) in
+theorem Algebra.IsAlgebraic.injective_tower_top (S) {A} [CommRing S] [NoZeroDivisors S]
+    [Algebra R S] [alg : Algebra.IsAlgebraic R S] [Ring A] [Algebra R A] [Algebra S A]
+    [IsScalarTower R S A] (inj : Injective (algebraMap R A)) : Injective (algebraMap S A) := by
```

Comment:
@ocfnash pointed out that in many instances, `Injective (algebraMap _ _)` can be replaced by `FaithfulSMul`. Is this an example of such an instance?

Reply:
They are always interchangeable, see [faithfulSMul_iff_algebraMap_injective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Basic.html#faithfulSMul_iff_algebraMap_injective). I don't know whether there is a plan to globally replace `Injective (algebraMap ..)` by `FaithfulSMul ..`; searching for `Injective (algebraMap` returns 132 results in 60 files. Should I try?

Reply:
In this case I'm using `Injective` instead of `FaithfulSMul` because [AlgebraicIndependent.algebraMap_injective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/AlgebraicIndependent/Basic.html#AlgebraicIndependent.algebraMap_injective) and [algebraicIndependent_empty_type_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/AlgebraicIndependent/Basic.html#algebraicIndependent_empty_type_iff) currently use `Injective`.

Reply:
I'm happy with merging this as is. But otoh, if you use `FaithfulSMul` then it could be an instance, right?

Reply:
Maybe we should have both?

Reply:
The conclusion doesn't mention R so it can't be an instance ... but I'm happy to add the theorem.